### PR TITLE
Add BOP (blue/orange/purple) colormaps

### PR DESCRIPTION
**Pull request by zeroth**
# Description
Adds new BOP [blue, orange, purple] colormaps based on https://github.com/cleterrier/ChrisLUTs

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

- [ x] the test_colormaps.py 

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/utils/colormaps/bop_colors.py [hunk 1]
```diff
@@ -0,0 +1,784 @@
+"""This module contains the color map dictionaries for BOP Lookup table take
+from https://github.com/cleterrier/ChrisLUTs to make it compatable with napari's
+colormap classes alll the values in the color map are normalized (divid by 255).
```
Comment (jni) [inline napari/utils/colormaps/bop_colors.py]:
```suggestion
"""This module contains the colormap dictionaries for BOP lookup tables taken
from https://github.com/cleterrier/ChrisLUTs. To make them compatible with napari's
colormap classes, all the values in the color map are normalized to the range [0, 1].
```

File: napari/utils/colormaps/bop_colors.py [hunk 2]
```diff
@@ -0,0 +1,784 @@
...
+    [0.5607843137254902, 0.12156862745098039, 0.5607843137254902],
+    [0.5647058823529412, 0.12156862745098039, 0.5647058823529412],
+    [0.5686274509803921, 0.12156862745098039, 0.5686274509803921],
+    [0.5686274509803921, 0.12156862745098039, 0.5686274509803921],
+    [0.5725490196078431, 0.12156862745098039, 0.5725490196078431],
+    [0.5725490196078431, 0.12156862745098039, 0.5725490196078431],
+    [0.5803921568627451, 0.12549019607843137, 0.5803921568627451],
+]
+
+bopd = {"BOPBlue": bop_blue, "BOPOrange": bop_orange, "BOPPurple": bop_purple}
```
Comment (jni) [inline napari/utils/colormaps/bop_colors.py]: These will be rendered by the UI, so they should be a bit nicer to read. Additionally, we typically use all lowercase. I suggest:

```suggestion
bopd = {"bop blue": bop_blue, "bop orange": bop_orange, "bop purple": bop_purple}
```

**Review by jni — COMMENTED** [review]
@zeroth thanks! I've made some very minor comments. Additionally, on my machine when I select one of these colormaps, the text on the colormap dropdown box changes to "custom" rather than the selected name. I don't actually know why that is! But we should figure it out before merging.

@napari/core-devs for context, these colormaps were requested by @zeroth's advisor Senthil, who is colorblind. I really liked BOP when I first saw it and I think improving our accessibility is a big deal!

**Comment by jni** [comment]
btw @zeroth could you follow the instructions in our [contributing guide](https://napari.org/docs/dev/developers/CONTRIBUTING.html) to set up your development environment to run Black and Flake8 locally before each commit?

File: napari/utils/colormaps/colormap_utils.py [hunk 3]
```diff
@@ -39,6 +40,8 @@
     for name, color in zip(primary_color_names, primary_colors)
 }
 
+# dictionay for bop colormap objects
+BOP_COLORMAPS = {k: Colormap(v) for k, v in bopd.items()}
```
Comment (kevinyamauchi) [inline napari/utils/colormaps/colormap_utils.py]: We need to pass the name of the colormap to the `Colormap` init method ([the default value is `custom`](https://github.com/napari/napari/blob/0055113d7ffc68e06c032f9cdf22ed2fbf9db0c6/napari/utils/colormaps/colormap.py#L24-L44) ). I think this is why the colormap name gets set to `custom` when selecting it in the GUI.

```suggestion
BOP_COLORMAPS = {k: Colormap(v, name=k) for k, v in bopd.items()}
```

**Review by kevinyamauchi — COMMENTED** [review]
Thanks, @zeroth ! I haven't taken a full look, but I took it for a spin and it looks nice! I suggested a fix below for colormap name when selecting them from the GUI.

I agree with @jni that is important that we aim to make napari usable for everyone!

**Comment by zeroth** [comment]
@jni for some reason on windows 10 pre-commit is failing, I fixed it (just moved to WSL ;)  ).
Update the code with your suggestions. 
@kevinyamauchi Cool, you liked it. Thanks for the tip updated the code please have a look.

File: napari/utils/colormaps/bop_colors.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
-"""This module contains the color map dictionaries for BOP Lookup table take
-from https://github.com/cleterrier/ChrisLUTs to make it compatable with napari's
-colormap classes alll the values in the color map are normalized (divid by 255).
+"""This module contains the colormap dictionaries for BOP lookup tables take
```
Comment (jni) [inline napari/utils/colormaps/bop_colors.py]:
```suggestion
"""This module contains the colormap dictionaries for BOP lookup tables taken
```

Comment (jni) [reply napari/utils/colormaps/bop_colors.py]: Just one final bit!

**Review by jni — APPROVED** [review]
:tada:

**Review by kevinyamauchi — APPROVED** [review]
Looks great to me! Thanks, @zeroth !

**Pull request merged**